scripts/prepare_dolly.py

```diff
@@ -0,0 +1,133 @@
+"""Implementation derived from https://github.com/tloen/alpaca-lora"""
+import sys
+from pathlib import Path
+
+# support running without installing as a package
+wd = Path(__file__).parent.parent.resolve()
+sys.path.append(str(wd))
+
+import torch
+import requests
+import json
+from torch.utils.data import random_split
+from lit_llama.tokenizer import Tokenizer
+from tqdm import tqdm
+
+
+DATA_FILE = "https://raw.githubusercontent.com/databrickslabs/dolly/master/data/databricks-dolly-15k.jsonl"
+DATA_FILE_NAME = "dolly_data_cleaned.json"
+IGNORE_INDEX = -1
+
+
+def prepare(
+    destination_path: Path = Path("data/dolly"), 
+    tokenizer_path: Path = Path("checkpoints/lit-llama/tokenizer.model"),
+    test_split_size: int = 2000,
+    max_seq_length: int = 256,
+    seed: int = 42,
+    mask_inputs: bool = False,  # as in alpaca-lora
+) -> None:
+    """Prepare the Alpaca dataset for instruction tuning.
+    
+    The output is a training and validation dataset saved as `train.pt` and `val.pt`,
+    which stores the preprocessed and tokenized prompts and labels.
+    """
+    
+    destination_path.mkdir(parents=True, exist_ok=True)
+    file_path = destination_path / DATA_FILE_NAME
+    download(file_path)
+
+    # TODO: If we don't have the Meta weights, where do we get the tokenizer from?
```

Comment:
This is a good question.
We should probably distribute the tokenizer model ourselves in the future, to make this less painful. It's 220k gzipped, and I don't think anyone would complain, no real IP in there.

Reply:
Not for this PR of course